core/src/main/java/org/apache/accumulo/core/conf/AccumuloConfiguration.java

```diff
@@ -209,13 +209,19 @@ static public long getMemoryInBytes(String str) {
         case 'B':
           multiplier = 0;
           break;
+        case '%':
```

Comment:
> Even if we limit percentage, users can still misconfigure things by setting fixed memory values that are too high or low for that property.

Thats true.  Using types just narrows the set of possible mistakes, it does not eliminate mistakes.